### PR TITLE
[FIX] Fixes deeplinks on Android 12 and other deeplinks minor fixes

### LINKED ISSUE #3395: Android deep link not opening if app is already open
**Describe the bug**
_A clear and concise description of what the bug is_
https://www.youtube.com/watch?v=lBHlW-7DV6E

**Screenshots**
_If applicable, add screenshots or links to help explain your problem_
https://www.youtube.com/watch?v=lBHlW-7DV6E

**To Reproduce**
_Steps to reproduce the behavior_
1. Open mobile browser (chrome) with deep link 1 (google)
2. This opens metamsk app > metamask browser > google  :thumbsup:
3. BUT if i go back to mobile browser (chrome) and open link 2 (youtube)
4. It will open metamask app > metamask browser > stays on google :thumbsdown:

**Expected behavior**
_A clear and concise description of what you expected to happen_
Expected to open link and redirect to the correct page.

**Smartphone (please complete the following information):**
 - Device: Google Pixel 3a
 - OS: Android
 - App Version [e.g. 1.0.0] - v3.6.0

-------------------------------------------------------------
_to be added after bug submission by internal support / PM_
**Severity**

- How critical is the impact of this bug on a user?
- Add stats if available on % of customers impacted
- Is this visible to all users?
- Is this tech debt?

### LINKED ISSUE #3525: Open deep link duplicate 5 or 9 tab
**Describe the bug**
im open deep link but metamask duplicate 5 tab or 9 tab, should be open 1 tab only

**Screenshots**
Screenshot_2021-12-27-17-52-01-145_io metamask

Screenshot_2021-12-27-17-52-48-315_io metamask

**To Reproduce**
1. Generate deep link from [](https://metamask.github.io/metamask-deeplinks/)
2. Open deep link

**Smartphone (please complete the following information):**
 - Device: Android / IOS
 - OS: Android 11 / IOS 15
 - App Version: 3.8.0

### LINKED ISSUE #3534: External App deep linking broken on Android 12
**Describe the bug**
On android devices with OS 12 installed, deep links for the dapp browser only open MetaMask and not the dapp browser.

**Screenshots**
NA

**To Reproduce**
_Steps to reproduce the behavior_
1. Load this URL into the browser https://southern-bush-399.notion.site/DeepLink-Test-e9b159d196424926809848705ffa015f 
2. Select any of the links 
3. The link opens MetaMask but does not load the page in the MM browser

**Expected behavior**
Expect the MM browser to be loaded with the deep link site

**Smartphone (please complete the following information):**
 - Device: [Pixel 5]
 - OS: [12]
 - App Version [3.8.0]

-------------------------------------------------------------
_to be added after bug submission by internal support / PM_
**Severity**

- How critical is the impact of this bug on a user?
- Add stats if available on % of customers impacted
- Is this visible to all users?
- Is this tech debt?

### PR DESCRIPTION
**Pull request by andreahaku**
**Description**
Fixes deeplinks on Android 12 and other deeplinks minor fixes like multiple tabs opening when tapping deep links from other apps and other minor issues with WalletConnect deep links.

**Checklist**

* [x] There is a related GitHub issue
* [ ] Tests are included if applicable
* [ ] Any added code is fully documented

**Issue**

Resolves #3534
Resolves #3395
Resolves #3525

### CODE REVIEW

**Review by blackdevelopa — APPROVED** [review]

File: android/app/src/main/AndroidManifest.xml [hunk 1]
```diff
@@ -51,10 +51,14 @@
 				<action android:name="android.intent.action.VIEW" />
 				<category android:name="android.intent.category.DEFAULT" />
 				<category android:name="android.intent.category.BROWSABLE" />
-				<data android:scheme="https" android:host="metamask.app.link" />
-				<data android:scheme="https" android:host="metamask-alternate.app.link" />
-				<data android:scheme="https" android:host="metamask.test-app.link" />
-				<data android:scheme="https" android:host="metamask-alternate.test-app.link" />
+				<data android:scheme="https" />
+				<data android:scheme="https" />
```
Comment (Cal-L) [inline android/app/src/main/AndroidManifest.xml:55]: These seem duplicate.

File: app/core/DeeplinkManager.js [hunk 2]
```diff
@@ -147,12 +147,13 @@ class DeeplinkManager {
...
 		const urlObj = new URL(url.replace('dapp/https://', 'dapp/').replace('dapp/http://', 'dapp/'));
 		let params;
+		let wcCleanUrl;
 
 		if (urlObj.query.length) {
 			try {
 				params = qs.parse(urlObj.query.substring(1));
 			} catch (e) {
-				Alert.alert(strings('deeplink.invalid'), e.toString());
+				if (e) Alert.alert(strings('deeplink.invalid'), e.toString());
```
Comment (Cal-L) [inline app/core/DeeplinkManager.js:156]: Is there a scenario where error is undefined?

File: android/app/src/main/AndroidManifest.xml (hunk 1 above)
Comment (andreahaku) [reply android/app/src/main/AndroidManifest.xml:55]: It’s not duplicated. Android 12 has a problem having the ‘scheme’ and ‘hosts’ on the same tag. So it needs to be divided into two tags. This is the solution that solves Android12 issues with deep links.

File: app/core/DeeplinkManager.js (hunk 2 above)
Comment (andreahaku) [reply app/core/DeeplinkManager.js:156]: yes, if `urlObj.query` is only 1 char, the `substring(1)` is `undefined`. this is happening sometimes with WC. It doesn't really apply here as `wallet connect` refactoring is not yet included.

File: app/components/Nav/App/index.js [hunk 3]
```diff
@@ -167,10 +165,7 @@ const App = ({ userLoggedIn }) => {
 			frequentRpcList,
 			dispatch,
 		});
-
-		unsubscribeFromBranch.current = branchSubscriber.subscribe();
-		return () => unsubscribeFromBranch.current?.();
-	}, [branchSubscriber, frequentRpcList, dispatch]);
+	}, [dispatch, frequentRpcList]);
```
Comment (gantunesr) [inline app/components/Nav/App/index.js:168]: Do you know why this wasn't part of the dependency array before?

File: android/app/src/main/AndroidManifest.xml (hunk 1 above)
Comment (gantunesr) [reply android/app/src/main/AndroidManifest.xml:55]: Is it possible to add a comment explaining this? I can see a lot of people questioning this in the future

Comment (andreahaku) [reply android/app/src/main/AndroidManifest.xml:55]: Found the link to Google documentation about it :)
https://developer.android.com/training/app-links/verify-site-associations#multi-subdomain
Adding it as a comment. Thanks @gantunesr

**Review by Cal-L — APPROVED** [review]
LGTM

**Review by cortisiko — APPROVED** [review]
🌮 🌮 🌮

**Pull request merged**